### ISSUE TITLE
Bamtools filter doesn't allow for filtering out specific chromosomes. 

### ISSUE THREAD

**Issue by zfrenchee** [galaxyproject/tools-devteam#348]
There are two BAM file filtering utilities that devteam maintains: `Filter SAM or BAM, output SAM or BAM` which is also called `samtools_filter2` I believe, and `bamtools_filter`. The latter always gives me this error:

```
terminate called after throwing an instance of 'std::out_of_range'
  what():  vector::_M_range_check
Aborted (core dumped)
[W::bam_hdr_read] EOF marker is absent. The input is probably truncated.
```

Which I would be prepared to believe, but the first of those two tools doesn't give me an error. 

What I'm trying to do is remove mitochondrial reads from a BAM file. `samtools_filter2` requires that I enumerate all the `chr` that I want to keep, while `bamtools_filter` lets me select ones that I want to throw away. The latter of those is much more convenient, since I want to keep _everything except_ chrM. 

Is there any chance that `bamtools_filter` is broken, and if so, could it be fixed? Otherwise, is there a way of specifying `chr`s that you want to throw away with 'samtools_filter2` that isn't documented? 

Thanks!

**Comment by zfrenchee** [comment]
One way of solving this would be to add a -U option, the way `samtools view` does, to the `samtools_filter2` tool.

**Comment by zfrenchee** [comment]
When I use [Bamtools filter](https://github.com/galaxyproject/tools-devteam/blob/0037a546e17ed2c32c9bf662aec292e31ac5e653/tool_collections/bamtools/bamtools_filter/bamtools-filter.xml) to filter out low quality reads, it doesnt give me the out of range error, but when I ask it to filter anything relating to chromosomes, it _does_. I think it might be broken. I see that it's a fairly new tool, at a low version.

**Comment by zfrenchee** [comment]
Solved in #351
